### PR TITLE
drivers: bt_nxp: delete non monolithic support on RW

### PR DESCRIPTION
**Pull request by yeaissa**
Current code does no longer support the non monolithic feature.
Hence, removing unnecessary code.

### CODE REVIEW

**Comment by yeaissa** [comment]
@axelnxp, please check

**Review by axelnxp — APPROVED** [review]

**Comment by decsny** [comment]
should the CONFIG_NXP_MONOLITHIC_BT just be deleted then?

**Comment by butok** [comment]
> should the CONFIG_NXP_MONOLITHIC_BT just be deleted then?

Agree. Zephyr platforms now do not support non-monolithic WiFi and BT.
There is no point in having user configuration parameters that cannot be changed.

**Comment by yeaissa** [comment]
> > should the CONFIG_NXP_MONOLITHIC_BT just be deleted then?
> 
> Agree. Zephyr platforms now do not support non-monolithic WiFi and BT. There is no point in having user configuration parameters that cannot be changed.

This Kconfig is still usefull, and it's used for the following cases:
- A soc could enable this nxp KConfig to mention the support of this feature. For example RW61x soc supports this and not MCXW71.
- To have some CI coverage for BLE on RW61x. CI on Zephyr doesn't fetch the blobs and the monolithic feature needs them. Not fetching the nxp blobs will lead to build failures for BLE apps. This is why we use CONFIG_NXP_MONOLITHIC_BT=n on the sample.yaml (used by twister) so that the build doesn't try to link the binary blob on the final image.

@axelnxp what is your POV on this ?

**Comment by axelnxp** [comment]
> > > should the CONFIG_NXP_MONOLITHIC_BT just be deleted then?
> > 
> > 
> > Agree. Zephyr platforms now do not support non-monolithic WiFi and BT. There is no point in having user configuration parameters that cannot be changed.
> 
> This Kconfig is still usefull for the following reasons:
> 
>     * A soc could enable this nxp KConfig to mention the support of this feature. For example RW61x soc supports this and not MCXW71.
> 
>     * To have some CI coverage for BLE on RW61x. CI on Zephyr doesn't fetch the blobs and the monolithic feature needs them. Not fetching the nxp blobs will lead to build failures for BLE apps. This is why we use CONFIG_NXP_MONOLITHIC_BT=n on the sample.yaml (used by twister) so that the build doesn't try to link the binary blob on the final image.
> 
> 
> @axelnxp what is your POV on this ?

I agree, especially for the second point.
@butok @decsny this PR mainly aims to remove selecting DT partition when the monolithic feature is not enabled, this behavior doesn't have sense on platforms like MCXW71. MCXW71 doesn't support the monolithic feature, and we don't want to use DT partition either, because it forces the use of mcuboot alongside since @decsny changes enabling mcuboot support on MCXW71. Or we would need overlays everywhere, and that's not what we want to do.
So basically, this `select` is not relevant anymore, but the monolithic kconfig still is.

**Review by butok — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

**Review by decsny — CHANGES_REQUESTED** [review]
> > > should the CONFIG_NXP_MONOLITHIC_BT just be deleted then?
> > 
> > 
> > Agree. Zephyr platforms now do not support non-monolithic WiFi and BT. There is no point in having user configuration parameters that cannot be changed.
> 
> This Kconfig is still usefull for the following reasons:
> 
>     * A soc could enable this nxp KConfig to mention the support of this feature. For example RW61x soc supports this and not MCXW71.
> 
>     * To have some CI coverage for BLE on RW61x. CI on Zephyr doesn't fetch the blobs and the monolithic feature needs them. Not fetching the nxp blobs will lead to build failures for BLE apps. This is why we use CONFIG_NXP_MONOLITHIC_BT=n on the sample.yaml (used by twister) so that the build doesn't try to link the binary blob on the final image.
> 
> 
> @axelnxp what is your POV on this ?

in this case the following things appear to me to need to change if I understand you correctly:
- the samples for RW which set this kconfig to `n` should be set to build only since ci will not flash the radio binary, it will not actually run
- the description of the kconfig is now wrong if its being used to indicate support instead of configure something like youre saying above. which, if it's not, then why are we changing it from default y to explicit n in tests? it seems like it is for configuring still, but you just make it not work when set to n, >>
- probably the appropriate thing to do is to make a new kconfig to indicate support set by soc, and then that kconfig is a dependency of this one
- the line you are deleting should probably actually be moved to rw soc kconfig

**Comment by yeaissa** [comment]
> > > > should the CONFIG_NXP_MONOLITHIC_BT just be deleted then?
> > > 
> > > 
> > > Agree. Zephyr platforms now do not support non-monolithic WiFi and BT. There is no point in having user configuration parameters that cannot be changed.
> > 
> > 
> > This Kconfig is still usefull for the following reasons:
> > ```
> > * A soc could enable this nxp KConfig to mention the support of this feature. For example RW61x soc supports this and not MCXW71.
> > 
> > * To have some CI coverage for BLE on RW61x. CI on Zephyr doesn't fetch the blobs and the monolithic feature needs them. Not fetching the nxp blobs will lead to build failures for BLE apps. This is why we use CONFIG_NXP_MONOLITHIC_BT=n on the sample.yaml (used by twister) so that the build doesn't try to link the binary blob on the final image.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @axelnxp what is your POV on this ?
> 
> in this case the following things appear to me to need to change if I understand you correctly:
> 
> * the samples for RW which set this kconfig to `n` should be set to build only since ci will not flash the radio binary, it will not actually run
> * the description of the kconfig is now wrong if its being used to indicate support instead of configure something like youre saying above. which, if it's not, then why are we changing it from default y to explicit n in tests? it seems like it is for configuring still, but you just make it not work when set to n, >>
> * probably the appropriate thing to do is to make a new kconfig to indicate support set by soc, and then that kconfig is a dependency of this one
> * the line you are deleting should probably actually be moved to rw soc kconfig

@decsny, 
Updated, does it look good for you now?

**Review by axelnxp — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]
